### PR TITLE
Publish CSVs to Google Drive

### PR DESCRIPTION
**Pull request by nodanaonlyzuul**
Hey @thisisstephenbetts 

Here's a PR from @holingpoon & I.

There's a new flag to the CLI.
It's `--google-parent-id` or `-g` for short. "Parent" is what google drive calls folders.
I've added your work address to the folder with the id "1JMX2gLKSfW39oSj9JWZ407MXbiK1kF0l".

If you run the script with `-g 1JMX2gLKSfW39oSj9JWZ407MXbiK1kF0l` and `-o google-sheets` you'll see the report show up [in the folder](https://drive.google.com/drive/u/1/folders/1JMX2gLKSfW39oSj9JWZ407MXbiK1kF0l).

Also - `start_date`s and `end_date`s of 'yesterday' or 'today' will result in the appropriate YYYY-MM-DD in the filename.

I love you..OOPS WRONG WINDOW.
Sorry...I love you..OOPS AGAIN!

### CODE REVIEW

File: lib/report_runner.rb [hunk 1]
```diff
@@ -34,14 +36,28 @@ def generate_report
...
+      output: @output,
+      google_parent_id: @google_parent_id
     })
 
     report_generator.generate_report!
   end
 
 private
 
+  def has_google_parent_id?
```
Comment (thisisstephenbetts) [inline lib/report_runner.rb:48]: This name doesn't imply that this is only needed if `output` is `google-sheets`. Given this name, I'd expect it to fail if `google_parent_id` is missing, whether `output` is `google-sheets` or not.

Maybe `check_google_parent_id?`? That's got a bit more wiggle room. 

`has_google_parent_id_if_needed?` or `needs_and_has_google_parent_id?` is too clunky I feel.

File: lib/report_runner.rb [hunk 2]
```diff
@@ -34,14 +36,28 @@ def generate_report
...
+    if @output == "google-sheets"
```
Comment (thisisstephenbetts) [inline lib/report_runner.rb:49]: This really doesn't matter, but in the interests of cutting down the number of lines, could this be the following?
```
if @output == "google-sheets" and @google_parent_id.nil?
    @errors << "requires a drive_parent_id for google-sheets output"
    return false
end
return true
```

File: output_2017-10-26_2017-11-02.csv [hunk 3]
```diff
@@ -0,0 +1,65 @@
+Search Term,Searched Repo,Searched From,Total Searches,Total Clicks,CTR,Mean Ordinality
```
Comment (thisisstephenbetts) [inline output_2017-10-26_2017-11-02.csv:1]: Did you mean to add this?

File: spec/report_runner_spec.rb [hunk 4]
```diff
@@ -24,10 +24,16 @@
       expect(report_runner.errors).to include("is missing auth_file")
     end
 
-    it "will have an error without auth_file" do
-      report_runner = ReportRunner.new
+    it "will have an error if given 'google-sheets' as an output but no drive_parent_id" do
```
Comment (thisisstephenbetts) [inline spec/report_runner_spec.rb:27]: Shouldn't `drive_parent_id` be  `google_parent_id`?

File: spec/report_runner_spec.rb [hunk 5]
```diff
@@ -24,10 +24,16 @@
...
+      report_runner = ReportRunner.new(output: 'google-sheets')
+      report_runner.valid?
+      expect(report_runner).to_not be_valid
+      
+      expect(report_runner.errors).to include("requires a drive_parent_id for google-sheets output")
+
+      report_runner.google_parent_id = "nowihaveavalue"
```
Comment (thisisstephenbetts) [inline spec/report_runner_spec.rb:34]: Now you're adding a `google_parent_id` shouldn't the next line be valid?

**Review by thisisstephenbetts — APPROVED** [review]
Got a couple of questions about one of the tests, but if I'm wrong about those this is approved. Couple of other minor comments.

File: spec/report_runner_spec.rb (hunk 5 above)
Comment (thisisstephenbetts) [reply spec/report_runner_spec.rb:34]: Okay — you explained this. The Runner is still not valid, but it is for other reasons. I think maybe a comment saying it's still invalid may have helped me, but I should have read the code more carefully.

Comment (nodanaonlyzuul) [reply spec/report_runner_spec.rb:34]: We discussed this in person but for posterity - we agreed...it's still invalid but now its errors array doesn't include the message about `google_parent_id`

Comment (thisisstephenbetts) [reply spec/report_runner_spec.rb:34]: I was 15s ahead of you! :-D

File: output_2017-10-26_2017-11-02.csv (hunk 3 above)
Comment (nodanaonlyzuul) [reply output_2017-10-26_2017-11-02.csv:1]: Uff no - I absolutely didn't this has been fixed in master cc52267fa6d3877a38632e73f8f1d54e6c982d57.

File: lib/report_runner.rb (hunk 1 above)
Comment (nodanaonlyzuul) [reply lib/report_runner.rb:48]: I agree - changes in 98164f08086d416c3ca1b6a3b483ed9056c840c2.

File: lib/report_runner.rb (hunk 2 above)
Comment (nodanaonlyzuul) [reply lib/report_runner.rb:49]: Much cleaner - yes that didn't feel good at the time of writing. Also fixed in 98164f08086d416c3ca1b6a3b483ed9056c840c2.

**Pull request merged**